source/time.tex

```diff
@@ -8039,713 +8077,340 @@
 
-\rSec1[time.tod]{Class template \tcode{time_of_day}}
+\rSec1[time.hms]{Class template \tcode{hh_mm_ss}}
 
-\rSec2[time.tod.overview]{Overview}
-\indexlibrary{\idxcode{time_of_day}}
+\rSec2[time.hms.overview]{Overview}
+\indexlibrary{\idxcode{hms}}
 
 \begin{codeblock}
 namespace std::chrono {
-  template<class Duration> class time_of_day;
-
-  template<> class time_of_day<hours>;
-  template<> class time_of_day<minutes>;
-  template<> class time_of_day<seconds>;
-  template<class Rep, class Period> class time_of_day<duration<Rep, Period>>;
-}
-\end{codeblock}
-
-\pnum
-The \tcode{time_of_day} class template
-splits a \tcode{duration} representing the time elapsed since midnight
-into a ``broken down'' time of day such as
-$hours$:$minutes$:$seconds$.
-The \tcode{Duration} template parameter dictates
-the precision to which the time is broken down.
-\begin{note}
-This can vary from a coarse precision of hours
-to a very fine precision of nanoseconds.
-\end{note}
-A \tcode{time_of_day} object also tracks
-whether it should be output
-as a 12-hour time format or a 24-hour time format.
+  template <class Duration>
+  class hh_mm_ss
+  {
+    bool            is_neg;     // \expos
+    chrono::hours   h;          // \expos
+    chrono::minutes m;          // \expos
+    chrono::seconds s;          // \expos
+    precision       ss;         // \expos
 
-\pnum
-The primary \tcode{time_of_day} template is not defined.
-Four specializations are provided to handle four different
-levels of precision.
-
-\pnum
-Each specialization of \tcode{time_of_day} is a trivially copyable
-and standard-layout class type.
-
-\rSec2[time.tod.hours]{Hours precision}
-\indexlibrary{\idxcode{time_of_day<hours>}}
-
-\begin{codeblock}
-namespace std::chrono {
-  template<>
-  class time_of_day<hours> {
   public:
-    using precision = chrono::hours;
+    static unsigned constexpr fractional_width = @\seebelow@;
+    using precision                            = @\seebelow@;
 
-    time_of_day() = default;
-    constexpr explicit time_of_day(chrono::hours since_midnight) noexcept;
+    constexpr hh_mm_ss() noexcept : hh_mm_ss{Duration::zero()} {}
+    constexpr explicit hh_mm_ss(Duration d);
 
+    constexpr bool is_negative() const noexcept;
     constexpr chrono::hours hours() const noexcept;
+    constexpr chrono::minutes minutes() const noexcept;
+    constexpr chrono::seconds seconds() const noexcept;
+    constexpr precision subseconds() const noexcept;
 
-    constexpr explicit operator  precision()   const noexcept;
-    constexpr          precision to_duration() const noexcept;
-
-    constexpr void make24() noexcept;
-    constexpr void make12() noexcept;
+    constexpr explicit operator precision() const noexcept;
+    constexpr precision to_duration() const noexcept;
   };
+
+  template <class charT, class traits, class Duration>
+    basic_ostream<charT, traits>&
+      operator<<(basic_ostream<charT, traits>& os, hh_mm_ss<Duration> const& hms);
```

Comment:
Repetition of this non-member function already specified in the header synopsis should be removed.

Reply:
Fixed.